main.go

```diff
@@ -254,9 +306,17 @@ func _main() error {
 	if err != nil {
 		return err
 	}
+	projectID := projectTmp.Name()
+	if cache {
+		projectID, err = getFilesContentHash(files, projectTmp.Name())
```

Comment:
You could have this getFilesContentHash return another variable that indiciates if it copied from stdin to a tmp file.

That way you don't need that second loop through `files` without all the duplicated code you added.

It could be something like:

```
projectId, stdinTmpCopy, err = getFilesContentHash(files, projectTmp.Name())
if stdinTmpCopy != "" {
  files = append(files, stdinTmpCopy)
}
```